RELEASING.md errors

* prior to `make translations` you need to run `npm install`
* make translations errors

```
$ make translations
node data/update_locales

fs.js:432
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^
Error: ENOENT, no such file or directory './transifex.auth'
    at Object.fs.openSync (fs.js:432:18)
    at Object.fs.readFileSync (fs.js:286:15)
    at Object.<anonymous> (/home/pnorman/osm/iD/data/update_locales.js:23:26)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
make: *** [translations] Error 8
```
* make suggestions installs node bindings for osmium. this seems unnecessary
* `git checkout release && git reset --hard master` errors with `error: pathspec 'release' did not match any file(s) known to git.

--- a/RELEASING.md
+++ b/RELEASING.md
@@ -5,6 +5,12 @@
   (https://www.transifex.com/projects/p/id-editor/announcements/)
 - [ ] Notify TomH
 
+### Prep
+- [ ] If you don't have a `transifex.auth` file in the root of your iD checkout,
+      you'll need to create a Transifex account, ask @bhousel for admin rights
+      on the iD project, and then create this file with contents
+      like `{"user": "yourusername", "password": "*******"}`
+
 ### Update master branch
 - [ ] `git checkout master`
 - [ ] `make translations`

